src/connections.jl

```diff
@@ -193,14 +199,19 @@ end
 
 # AbstractLock primitives:
 # https://github.com/JuliaLang/julia/blob/master/base/condition.jl#L18
-Base.lock(conn::Connection) = lock(conn.lock)
-Base.unlock(conn::Connection) = unlock(conn.lock)
-Base.trylock(conn::Connection) = trylock(conn.lock)
-Base.islocked(conn::Connection) = islocked(conn.lock)
+Base.lock(conn::Connection) = acquire(conn.semaphore)
+Base.unlock(conn::Connection) = release(conn.semaphore)
+Base.islocked(conn::Connection) = conn.semaphore.curr_cnt >= conn.semaphore.sem_size
```

Comment:
We should raise an issue up stream that semaphore does not implement `islocked`

Reply:
https://github.com/JuliaLang/julia/pull/32299